eegnb/experiments/visual_n170/n170.py

```diff
@@ -17,107 +17,40 @@
-
-    # Setup graphics
-    mywin = visual.Window([1600, 900], monitor="testMonitor", units="deg", fullscr=True)
-
     faces = list(map(load_image, glob(os.path.join(FACE_HOUSE, "faces", "*_3.jpg"))))
     houses = list(map(load_image, glob(os.path.join(FACE_HOUSE, "houses", "*.3.jpg"))))
-    stim = [houses, faces]
-
-    # Show the instructions screen
-    show_instructions(duration)
-
-    if eeg:
-        if save_fn is None:  # If no save_fn passed, generate a new unnamed save file
-            random_id = random.randint(1000,10000)
-            save_fn = generate_save_fn(eeg.device_name, "visual_n170", random_id, random_id, "unnamed")
-            print(
-                f"No path for a save file was passed to the experiment. Saving data to {save_fn}"
-            )
-        eeg.start(save_fn, duration=record_duration + 5)
-
-    # Start EEG Stream, wait for signal to settle, and then pull timestamp for start point
-    start = time()
-
-    # Iterate through the events
-    for ii, trial in trials.iterrows():
-        # Inter trial interval
-        core.wait(iti + np.random.rand() * jitter)
-
-        # Select and display image
-        label = trials["image_type"].iloc[ii]
-        image = choice(faces if label == 1 else houses)
-        image.draw()
-
-        # Push sample
-        if eeg:
-            timestamp = time()
-            if eeg.backend == "muselsl":
-                marker = [markernames[label]]
-            else:
-                marker = markernames[label]
-            eeg.push_sample(marker=marker, timestamp=timestamp)
 
-        mywin.flip()
-
-        # offset
-        core.wait(soa)
-        mywin.flip()
-        if len(event.getKeys()) > 0 or (time() - start) > record_duration:
-            break
-
-        event.clearEvents()
-
-    # Cleanup
-    if eeg:
-        eeg.stop()
-
-    mywin.close()
-
-
-def show_instructions(duration):
-
-    instruction_text = """
-    Welcome to the N170 experiment! 
- 
-    Stay still, focus on the centre of the screen, and try not to blink. 
-
-    This block will run for %s seconds.
-
-    Press spacebar to continue. 
+    return [houses, faces]
     
-    """
-    instruction_text = instruction_text % duration
+def present_stimulus(trials, ii, eeg, markernames):
+   
+    label = trials["image_type"].iloc[ii]
+    image = choice(faces if label == 1 else houses)
+    image.draw()
 
-    # graphics
-    mywin = visual.Window([1600, 900], monitor="testMonitor", units="deg", fullscr=True)
-
-    mywin.mouseVisible = False
-
-    # Instructions
-    text = visual.TextStim(win=mywin, text=instruction_text, color=[-1, -1, -1])
-    text.draw()
-    mywin.flip()
-    event.waitKeys(keyList="space")
+    # Push sample
+    if eeg:
+        timestamp = time()
+        if eeg.backend == "muselsl":
+            marker = [markernames[label]]
+        else:
+            marker = markernames[label]
+        eeg.push_sample(marker=marker, timestamp=timestamp)
```

Comment:
@JohnGriffiths This muselsl-edgecase should prob be handled directly in `push_sample`, and not in calling code. (not related to this PR)

Reply:
Also the whole push sample part should be inside the experiment base class; in the majority of cases just comes  immediately after stim presentation and doesn't need to be repeated